packages/firestore/src/remote/watch_change.ts

```diff
@@ -535,14 +531,18 @@ export class WatchChangeAggregator {
       this.ensureDocumentTargetMapping(key).delete(targetId)
     );
 
-    if (updatedDocument) {
+    if (updatedDocument && !this.pendingDocumentUpdates.get(key)) {
```

Comment:
Hrm. Not sure why you added the `!this.pendingDocumentUpdates.get(key)` check.  If it's correct, please add a comment that convinces me.